### ISSUE TITLE
Exceptions get thrown after return

### ISSUE THREAD

**Issue by zdimension** [CosmosOS/Cosmos#376]
In one of my projects, there is a function like this:

``` csharp
public FileSystem GetFS(string path)
{
    foreach (var t in MountedFileSystems)
    {
        if (condition)
        {
            return t.FS;
        }
    }
    throw new Exception("The mount point '" + path + "' doesn't exist");
}
```

Even if the condition is met, and the `return` statement reached, it still throws the exception. The workaround I'm using is:

``` csharp
public FileSystem GetFS(string path)
{
    FileSystem res = null;
    foreach (var t in MountedFileSystems)
    {
        if (condition)
        {
            res = t.FS;
            break;
        }
    }
    if (res == null)
        throw new Exception("The mount point '" + path + "' doesn't exist");
    return res;
}

```

**Comment by mterwoord** [comment]
This is on devkit or userkit?

**Comment by zdimension** [comment]
It's the latest devkit.

**Comment by jp2masa** [comment]
I tried this code (on a devkit of a very recent commit) and when it returns it doesn't throw the exception. Was the bug fixed or is the problem with the FileSystem?

    public class Kernel : Sys.Kernel
    {
        protected override void BeforeRun()
        {
          Console.Clear();
          Console.WriteLine("Cosmos booted successfully. Type a line of text to get it echoed back.");
        }

        protected override void Run()
        {
          Console.Write("Input: ");
          var input = Console.ReadLine();
          Console.Write("Text typed: ");
          Console.WriteLine(GetString(input));
        }

        public string GetString(string str)
        {
          foreach(char c in str)
          {
            if (c.ToString() == "0")
              return str;
          }

          throw new Exception("String must have less than 10 characters.");
        }
      }

**Comment by mterwoord** [comment]
Can be fixed, but as you're not using filesystem, it can't be related to that.

so you're saying on latest devkit, it's _not_ an issue anymore?

**Comment by jp2masa** [comment]
I didn't use filesystem and it runs fine(doesn't throw exception) so if there is a issue it can be related to the filesystem.

**Comment by mterwoord** [comment]
So if this issue is solved now, can you please close it? Thanks!

**Comment by jp2masa** [comment]
It wasn't me who opened this issue, I was just trying to help fixing it.

**Comment by mterwoord** [comment]
Shoot, sorry..
@zdimension Can you please retry with latest master?

**Comment by zdimension** [comment]
The `FileSystem` here isn't related to Cosmos'. I'm gonna try with the latest devkit.

**Comment by jp2masa** [comment]
I think this issue has been fixed by #507
